### PR TITLE
feautre(nyz): add mario demo

### PR DESCRIPTION
**Pull request by PaParaZz1**
## Description
gym-super-mario-bros==7.3.0

Screen Shot 2022-08-13 at 10 43 32 PM

https://user-images.githubusercontent.com/33195032/184481953-5bdbb756-6cf4-492c-9d4e-ff5ccc34d4c6.mp4

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/envs/env_manager/base_env_manager.py [hunk 1]
```diff
@@ -464,6 +465,7 @@ def step(self, actions: List[tnp.ndarray]) -> List[tnp.ndarray]:
             # make the type and content of key as similar as identifier,
             # in order to call them as attribute (e.g. timestep.xxx), such as ``TimeLimit.truncated`` in cartpole info
             info = make_key_as_identifier(info)
+            info = remove_illegal_item(info)
```
Comment (zjowowen) [inline ding/envs/env_manager/base_env_manager.py:468]: Why should we have to remove string in info?

File: ding/envs/env/ding_env_wrapper.py [hunk 2]
```diff
@@ -127,16 +126,17 @@ def random_action(self) -> np.ndarray:
...
+        # wrapper_cfgs: Union[str, List]
+        wrapper_cfgs = self._cfg.get('env_wrapper', 'default')
         if isinstance(wrapper_cfgs, str):
             wrapper_cfgs = get_default_wrappers(wrapper_cfgs, self._cfg.get('env_id', None))
         self._wrapper_cfgs = wrapper_cfgs
-        for wrapper_cfg in self._wrapper_cfgs:
-            if wrapper_cfg.get('disable', False):
-                continue
-            self._env = create_env_wrapper(self._env, wrapper_cfg)
+        for wrapper in self._wrapper_cfgs:
```
Comment (zjowowen) [inline ding/envs/env/ding_env_wrapper.py:134]: If "wrapper" is from "self._wrapper_cfgs", which is from "wrapper_cfgs", which is from "get_default_wrappers" or "self._cfg.get('env_wrapper', 'default')", both of which would return list of string.
So what is the cases when wrapper is callable?

File: dizoo/mario/mario_dqn_main.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        for i in range(cfg.policy.learn.update_per_collect):
+            train_data = replay_buffer.sample(learner.policy.get_attribute('batch_size'), learner.train_iter)
+            if train_data is None:
+                break
+            learner.train(train_data, collector.envstep)
+    # evaluate
+    evaluator_env = BaseEnvManager(
+        env_fn=[wrapped_mario_env for _ in range(evaluator_env_num)], cfg=cfg.env.manager
+    )
+    evaluator_env.enable_save_replay(cfg.env.replay_path)  # switch save replay interface
```
Comment (zjowowen) [inline dizoo/mario/mario_dqn_main.py:100]: Is there any way to enable the replay saving, while not having to construct a new environment manager?

File: ding/envs/env_manager/base_env_manager.py (hunk 1 above)
Comment (PaParaZz1) [reply ding/envs/env_manager/base_env_manager.py:468]: it(string) is not compatible with Tensor

File: dizoo/mario/mario_dqn_main.py (hunk 3 above)
Comment (PaParaZz1) [reply dizoo/mario/mario_dqn_main.py:100]: no, we need subprocess to speed up training, while only base env manager can enable save replay

File: setup.py [hunk 4]
```diff
@@ -168,6 +168,9 @@
         'sokoban': [
             'gym-sokoban',
         ],
+        'mario': [
+            'gym-super-mario-bros==7.3.0',
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply setup.py:172]: modified it

**Pull request merged**